src/components/switchTheme/useSwitchTheme.tsx

```diff
@@ -0,0 +1,56 @@
+import { createContext, ReactNode, useCallback, useContext, useEffect, useState } from 'react';
+
+interface ThemeContextType {
+  theme: 'light' | 'dark';
+  toggleTheme: () => void;
+}
+
+interface IProps {
+  children: ReactNode;
+}
+
+export const ThemeContext = createContext<ThemeContextType | null>(null);
+
+const ThemeContextProvider = ({ children }: IProps) => {
+  const getInitialTheme = () => {
+    const savedTheme = localStorage.getItem('theme');
+    if (savedTheme) {
+      return savedTheme as 'light' | 'dark';
+    }
+
+    const prefersDark = window.matchMedia('(prefers-color-scheme: dark)').matches;
+    return prefersDark ? 'dark' : 'light';
+  };
+
+  const [theme, setTheme] = useState<'light' | 'dark'>(getInitialTheme);
+
+  const toggleTheme = useCallback(() => {
+    setTheme((prev) => {
+      const newTheme = prev === 'light' ? 'dark' : 'light';
+      document.documentElement.setAttribute('data-theme', newTheme);
+      localStorage.setItem('theme', newTheme);
+      return newTheme;
+    });
+  }, []);
+
+  useEffect(() => {
+    document.documentElement.setAttribute('data-theme', theme);
+  }, [theme]);
```

Comment:
why do you use useEffect here?

Reply:
frankly speaking we dont need useCallback here, as toggleTheme doesn't depend on any external variables that could change.

Reply:
Because without useEffect, there would be no way to update the data-theme attribute when the theme changes. Consequently, the theme switching mechanism wouldn't work as expected.

Reply:
If I will remove useEffect , I will face the issue when on the first render no theme adding and only adds when I click on burger menu. We had such problem before. Do you remember routes?

Reply:
I can't agree with both statements. You should understand components life-cycle and how useCallback-useMemo-useEffect hook work.

Reply:
If I will leave like this it will work but it will not define user's windows (light-dark) theme settings correctly
```
document.documentElement.setAttribute('data-theme', theme)
```

Without useEffect, it wouldn't be dynamically applying the theme to the document after detecting the OS theme preference.